### ISSUE TITLE
MDAnalysis unavailable error

### ISSUE THREAD

**Issue by soviete** [BradyAJohnston/MolecularNodes#24]
I keep getting "Error: MDAnalysis Unavaiable" despite having MDanalysis installed.
I just  followed the instructions, and installed:

- Blender 3.1.2 (ok)
- Molecular Nodes 0.5.19 (ok), 
- Atomium (ok), 
- `python3 -m pip install MDAnalysis` (ok)

However, I keep getting the error "MDAnalysis Unavaiable" (I obtain the path from `import MDAnalysis as mda; mda.__path__`)

**Comment by BradyAJohnston** [comment]
Can you share a screenshot of the preferences panel, and the path for for the MDA installation? Any more info about your python installation etc.

**Comment by soviete** [comment]
Sure, thanks for the prompt response. The path is ~/.local/lib/python3.10/site-packages and I am sharing a screenshot of the preferences panel

![Screenshot from 2022-05-23 12-35-03](https://user-images.githubusercontent.com/7021628/169801760-d3e0a42f-b4f0-40c5-aafd-0cdac73b5e93.png)

**Comment by BradyAJohnston** [comment]
Hmm there isn't anything immediately wrong that I can see. I assume something just isn't working about the path. 

Can you try running this inside of Blender's script editor and seeing if it works?

```python
import sys
import os
import site

# path to python.exe
python_exe = os.path.realpath(sys.executable)


def verify_user_sitepackages(package_location):

    if os.path.exists(package_location) and package_location not in sys.path:
        sys.path.append(package_location)


verify_user_sitepackages(site.getusersitepackages())
verify_user_sitepackages('/.local/lib/python3.10/site-packages')


import MDAnalysis as mda
print("succesfully imported")
```

**Comment by BradyAJohnston** [comment]
Could you try also maybe adding another forward slash to the end of the path inside of the package directory? 

Blender seems to weird about wanting / not wanting those slashes at the end.

**Comment by soviete** [comment]
Output below:

Read prefs: /home/soviet/.config/blender/3.1/config/userpref.blend
Traceback (most recent call last):
  File "/Text", line 19, in <module>
  File "/home/soviet/.local/lib/python3.10/site-packages/MDAnalysis/__init__.py", line 207, in <module>
    from .coordinates.core import writer as Writer
  File "/home/soviet/.local/lib/python3.10/site-packages/MDAnalysis/coordinates/__init__.py", line 728, in <module>
    from . import chemfiles
  File "/home/soviet/.local/lib/python3.10/site-packages/MDAnalysis/coordinates/chemfiles.py", line 84, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'
Error: Python script failed, check the message in the system console

I double checked, and I do have the packaging module installed in python3.10 (i.e. 'Requirement already satisfied: packaging in /usr/lib/python3/dist-packages (21.3)')

I also tried adding another forward slash to the end of the path without success

**Comment by jmbuhr** [comment]
Same problem here

**Comment by BradyAJohnston** [comment]
Hmm - do you have more information about your systems & python installations? I can't replicate the error myself. Might be worth trying a different python installation if possible?

**Comment by jmbuhr** [comment]
@soviete you are also on linux, right? Debian-based? I'm on pop-os 22.04 personally

**Comment by BradyAJohnston** [comment]
From the error it seems like the path is working (it's calling MDAnalysis) but it's not initiating because of the 'lack' of `packaging`.

Can you try running this in your Blender script editor? It might be that your system python does have `packaging` but Blender's bundled python doesn't have it accessible as it's also not on path.

```python
import subprocess
import sys
import os
import site

# path to python.exe
python_exe = os.path.realpath(sys.executable)

# upgrade pip
subprocess.call([python_exe, "-m", "ensurepip"])
subprocess.call([python_exe, "-m", "pip", "install",
                "--upgrade", "pip"], timeout=600)

# install required packages
subprocess.call([python_exe, "-m", "pip", "install", "packaging"], timeout=600)


def verify_user_sitepackages(package_location):
    if os.path.exists(package_location) and package_location not in sys.path:
        sys.path.append(package_location)


verify_user_sitepackages(site.getusersitepackages())

import MDAnalysis as mda
print("success!")
```

**Comment by soviete** [comment]
This worked, Brady, thanks a lot!

**Comment by jmbuhr** [comment]
This fixed it for me, too. On the manual installation of the blender linux version.

**Comment by jmbuhr** [comment]
hm, so the check for MDAnalysis new works, but the `Open` and `Import Frames` buttons on `Local File` and `MD Trajectory` remain grayed out with files selected to import.
Any idea for debugging this?

**Comment by BradyAJohnston** [comment]
The buttons will remain greyed out until Blender detects file that exist. For some reason Blender has a hard time with relative paths, but also seems to use them places for some reason (inconsistently). 

When selecting the path, ensure that relative path is unticked: 
<img width="1041" alt="image" src="https://user-images.githubusercontent.com/36021261/169821803-aae1e977-8f0d-4f1c-896e-fb69ca52899b.png">

**Comment by jmbuhr** [comment]
Thank you very much! I just tried it. Consider my mind blown!

**Comment by BradyAJohnston** [comment]
Glad to see it's working!

**Comment by HebaMohamed88** [comment]
Hi Brady, I have the same problem. I tried the above script it failed! The path is /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages
<img width="601" alt="Screen Shot 2022-05-27 at 2 46 37 pm" src="https://user-images.githubusercontent.com/106366999/170634272-31afb89d-b1b4-4e7d-b3b1-3b76e0916d84.png">

**Comment by BradyAJohnston** [comment]
Hi @HebaMohamed88 are you able to provide more detail about what you mean when 'it failed'? Can you show screenshots of specific error messages, and provide the version of Blender, version of Molecular Nodes and OS you are working on?

**Comment by HebaMohamed88** [comment]
Blender: 3.1.2
Molecular Nodes:0.5.20
OS:12.2.1
I installed python 3.10, python3 -m pip install MDAnalysis, import MDAnalysis as mda; mda.__path__
copied the path /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/
<img width="493" alt="Screen Shot 2022-05-27 at 3 18 20 pm" src="https://user-images.githubusercontent.com/106366999/170638227-900c5a23-ca34-431f-90b9-4888e4398f95.png">

**Comment by HebaMohamed88** [comment]
I tried to run the above script inside of Blender's script editor 
output
Traceback (most recent call last):
  File "<blender_console>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/MDAnalysis/__init__.py", line 211, in <module>
    from . import converters
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/MDAnalysis/converters/__init__.py", line 35, in <module>
    from . import ParmEd
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/MDAnalysis/converters/ParmEd.py", line 84, in <module>
    from ..coordinates import base
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/MDAnalysis/coordinates/__init__.py", line 750, in <module>
    from . import GSD
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/MDAnalysis/coordinates/GSD.py", line 55, in <module>
    import gsd.fl
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/gsd/fl.cpython-310-darwin.so, 0x0002): tried: '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/gsd/fl.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/lib/fl.cpython-310-darwin.so' (no such file)

**Comment by BradyAJohnston** [comment]
@HebaMohamed88 it seems like there is compatibility in terms of architecture for your python installation.

```
tried: '/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/gsd/fl.cpython-310-darwin.so'

(mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), 
'/usr/lib/fl.cpython-310-darwin.so' (no such file)
```

How did you install your system Python? Are you on an M1 mac or an intel Mac? 

Likely will need to try reinstalling your python, or maybe additionally reinstalling blender, to ensure everything is compatible with your system (`arm64` if M1, `x86_64` if an intel Mac).

**Comment by BradyAJohnston** [comment]
Additionally, as per your screenshot, you currently have Molecular Nodes 0.5.15 installed, not 0.5.20. I recommend installing the latest build to get access to bug fixes.

**Comment by HebaMohamed88** [comment]
Thanks Brady,

Im on M1 mac, installed Python 3 release from https://www.python.org/downloads/macos/
straightforward installation. Im trying to figure out the problem, I reinstalled both Python and blender with no advance.
Thanks for your time.
[https://www.python.org/static/opengraph-icon-200x200.png]<https://www.python.org/downloads/macos/>

Python Releases for macOS | Python.org<https://www.python.org/downloads/macos/>
The official home of the Python Programming Language
www.python.org


________________________________

**Comment by BradyAJohnston** [comment]
Can you try running this code inside of your python installation? Either as a script o line by line in the console. Might be that the MDAnalysis isn't installing properly.


```python
import MDAnalysis as mda 

pdb = mda.fetch_mmtf('4ozs')

print("Succesfully import mda")
```

**Comment by HebaMohamed88** [comment]
Thanks Brady, the issue was I installed Blender the Intel version not apple silicon. It works, thank you so much.

**Comment by ahenao** [comment]
Hi. Thanks Brady! I have had a similar problem. It was not able to locate fasteners package. I saw this was because it was in a different path (/usr/local/lib) as where I had MDAnalysis (/home/my_user/.local/...).

I have solved this by installing MDAnalysis inside a conda environment. I did this with miniconda3:

First I installed miniconda:
>> bash Miniconda3-latest-Linux-x86_64.sh

Then I created the environment:

>> conda create --name MDAnalysis
>> conda activate MDAnalysis
>> conda install -c conda-forge MDAnalysis

Then with >> conda list     I checked I had also fasteners inside the environment

Then FInally I passed the conda environment path to blender (/home/my_user/miniconda3/envs/MDAnalysis/) and it worked

**Comment by SanderEwen** [comment]
Hi Brady,

Have also an error in windows 11, when checking the MDAnalysis installation. I've downloaded and installed Python 3.10.5 (latest version), with Blender 3.2 and Molecular Nodes 0.8.1. I've installed MDAnalysis and found the path inside Python. When pasting the path inside the MDAnalysis path, I get an error: MDAnalysis Unavailable

I've tried:
- Upgrade pip to the newest version;
- Open blender as administrator;
- Restart the computer;
- Remove the final backslash;
- pasting your script of 23th May in the Blender script editor.

Blender script editor output gives back a ModuleNotFoundError: No module named 'MDAnaysis'. So in a way blender can't access the installed MDAnalysis. Do you have any ideas?

Tnx in advance!

Sander

**Comment by BradyAJohnston** [comment]
Hi @SanderEwen, thanks for reporting the error! Are you able to give a screenshot of / paste more of the error output from the python console when you try to click the 'Check MDAnalysis' button? Can you also share a screenshot of the properties panel where the path is pasted so I can see what it looks like.

**Comment by SanderEwen** [comment]
> Hi @SanderEwen, thanks for reporting the error! Are you able to give a screenshot of / paste more of the error output from the python console when you try to click the 'Check MDAnalysis' button? Can you also share a screenshot of the properties panel where the path is pasted so I can see what it looks like.

Thanks for your fast reply :) See your requests in my original post.

**Comment by BradyAJohnston** [comment]
Hi @SanderEwen, looking at the path in the settings, try removing the ` \MDAnalysis\ ` from the end of the path so it will be : 

```
C:\Users\**\**\*\site-packages
```

**Comment by SanderEwen** [comment]
> Hi @SanderEwen, looking at the path in the settings, try removing the `\MDAnalysis\` from the end of the path so it will be :
> 
> ```
> C:\Users\**\**\*\site-packages
> ```

Great! This action works! Thanks a lot Brady. Keep up the good work :-)

**Comment by orbeckst** [comment]
I just wanted to add that *removing the trailing slash from the package directory* when setting up the plugin made the difference for me:

* **not** working: ~`/Users/***/anaconda3/envs/mda_blender32_py310/lib/python3.10/site-packages/`~
* **working**: `/Users/***/anaconda3/envs/mda_blender32_py310/lib/python3.10/site-packages`

-----

Everything else below is just context:

## conda installation of MDAnalysis for Blender

I installed MDAnalysis with conda (on macOS x86) (if you don't have a conda environment you can set one up by [installing miniconda](https://docs.conda.io/en/latest/miniconda.html#installing)
```bash
conda create -c conda-forge  -n mda_blender32_py310 python=3.10 mdanalysis
```
Then find the the package directory as described in the installation, or do it from the command line with
```bash
# activate your new environment
conda activate mda_blender32_py310

# use Python to find the package directory location (with trailing slash removed...)
python -c "import os.path; import MDAnalysis; print(os.path.dirname(MDAnalysis.__path__[0]))"

# optionally: deactivate the environment
conda deactivate
```

## Error message when loading a trajectory
**With trailing slash:** 

The plugin check for MDAnalysis succeeds ("MDAnalysis available") but then when I try to *Import Frames* in the *Load MD Trajectory File* panel I get `UnboundLocalError: local variable 'mda' references before assignment`
![molecularnodes_mda_conda_error](https://user-images.githubusercontent.com/237980/177221303-aabc647c-d14c-4464-99a7-3ae068a4895a.png)
presumably because the try/except import in the plugin failed but continued anyway — maybe it would be better to fail hard?

**Comment by BradyAJohnston** [comment]
Hi @orbeckst thanks for the detailed info! I'll add your notes on creating a conda environment to the docs, they are very helpful. 

I'll also fix up the check so that yes it will fail hard instead. This is my first python project so I'm still getting my head around how things should be done inside of Python (R for life).

**Comment by ChenouZhang** [comment]
@orbeckst Thanks for the installation guide! That's really helpful and I got atomium and MDA linking to work(now both button are grey). When I was trying to load the pdb and xtc, I still got the following error. It seems to be something from blender
![blender-bug](https://user-images.githubusercontent.com/42550849/177223792-98dbd286-5d31-43cf-be93-ffde7c200c57.png)
.

**Comment by BradyAJohnston** [comment]
Hi @ChenouZhang - which version of Blender and which version of Molecular Nodes do you have? Seems like the python libraries are linked OK but that in the creation of some of the nodes there are issues.

**Comment by ChenouZhang** [comment]
@BradyAJohnston I was using Blender-3.2.0 + molecular_nodes_0.8.1. Maybe I should try an older version of molecular_nodes?

**Comment by BradyAJohnston** [comment]
@ChenouZhang will address in a new issue #35

**Comment by MiekeRoth** [comment]
Still struggling with the same problem. I get the message when running the script from May 23. I have Blender 3.2.1 and plugin version 0.9.1: 
Traceback (most recent call last):
  File "<blender_console>", line 1, in <module>
ModuleNotFoundError: No module named 'MDAnalysis'

Thing is, I installed MDanalysis with Anaconda. And as a result I get a string like this: 
/.conda/pkgs/mdanalysis-2.2.0-py39h2e25243_0/Lib/site-packages
If I make the string /.conda/pkgs/ it gives the same message

**Comment by BradyAJohnston** [comment]
Hi @MiekeRoth what OS are you working on? Are you able to provide a screenshot of the preferences for the addon, so I can see all of the paths etc. 

Can you be more specific about which script you have tried, and what the error was when you tried it? There were a couple of different scripts posted on that day.

**Comment by MiekeRoth** [comment]
Hi @BradyAJohnston,
I work on Windows 11. And here are the preferences:
![mdanalysis_](https://user-images.githubusercontent.com/77438786/179390841-93fb700a-55cd-40fd-a68d-64b8fb63dbeb.png)
the path I use is: C:\Users\xxxx\.conda\pkgs\mdanalysis-2.2.0-py39h2e25243_0\Lib\site-packages\
I basically tried every script of that day I must confess. In all iterations. with and without the end \. Also tried going back to C:\Users\xxxx\.conda\pkgs\ with the same iterations. 

The error I keep getting is what I wrote above:
Traceback (most recent call last):
File "<blender_console>", line 1, in
ModuleNotFoundError: No module named 'MDAnalysis'

**Comment by BradyAJohnston** [comment]
Are you able to test out if the MDAnalysis installation is working? By importing it in a python session in the environment, and trying something like the following: 

```python
import MDAnalysis as mda
mda.fetch_mmtf('4ozs')
print("success!")
```

And see if that works? If it doesn't then it's an issue with the installation, if it does then it's still a path issue.

**Comment by MiekeRoth** [comment]
@BradyAJohnston I will try that later today (I hope). Need to finish some work before tomorrow. ;-) I will come back to it.

**Comment by MiekeRoth** [comment]
@BradyAJohnston I just copied the complete text. I'm getting the same error. Within the anaconda environment I also checked by 'conda list' and mdanalysis does appear. I'm searching for other sources on why it doesn't work, but until now haven't found anything that explains it. 


PYTHON INTERACTIVE CONSOLE 3.10.2 (main, Jan 27 2022, 08:34:43) [MSC v.1928 64 bit (AMD64)]

Builtin Modules:       bpy, bpy.data, bpy.ops, bpy.props, bpy.types, bpy.context, bpy.utils, bgl, blf, mathutils
Convenience Imports:   from mathutils import *; from math import *
Convenience Variables: C = bpy.context, D = bpy.data

>>> import MDAnalysis as mda
Traceback (most recent call last):
  File "<blender_console>", line 1, in <module>
ModuleNotFoundError: No module named 'MDAnalysis'

>>> mda.fetch_mmtf('4ozs')
Traceback (most recent call last):
  File "<blender_console>", line 1, in <module>
NameError: name 'mda' is not defined

>>> print("success!")
success!

>>>

**Comment by orbeckst** [comment]
Did you use the standard `python` interpreter that belongs to the conda environment to check? (I’m wondering because the “convenience imports do not look standard to me.) If you use an interpreter that’s no installed into the conda env then it will not know about modules that were installed there. 

If you created a new environment (like mda_blender32_py310 in my example above), did you switch to the environment first?

(apologies in advance, I’m on macOS or Linux so I don’t know exactly how things appear under Windows)

**Comment by RobertArbon** [comment]
Hi folks, I just thought I'd let you know how I got my python/blender setup working.  I'm on  Ubuntu 20.04.4 LTS; 'm using Blender 3.2.1 and MolecularNodes v0.10.4 (git SHA: 377dc073a72d406456c48cb98dc243e195486f2a). 

I followed the advice [here](https://stackoverflow.com/questions/70639689/how-to-use-the-anaconda-environment-on-blender) and did the following and created a conda environment: 

```
conda create --name blender python==3.10 mdanalysis atomium -c conda-forge -y
```
This will create an environment in (will be slightly different on your machine, depending on whether you're using miniconda/anaconda etc.)
```
/path/to/miniconda3/envs/blender
```
this folder will contain the following folders (amongst others): `bin`, `lib` and `include`.  

The python version you install must match the Blender python version.  You can inspect the python distro packaged with Blender to determine this.  Go to your Blender download folder (e.g., `/path/to/blender/blender-3.2.1-linux-x64/`). In there find the python executable in the `3.2/python/bin` subfolder.  Inspect the python executable (e.g,, `python3.10` means you should install python 3.10). 

Now rename the Blender python folder by navigating to  `/path/to/blender/blender-3.2.1-linux-x64/3.2/` and doing: 
```
mv python _python
```
Now Blender won't find the packaged distro. 

Now make a soft-link to the anaconda version of python

```
ln -s /path/to/miniconda3/envs/blender python
```

The final result should match: 
```
(blender) ➜  3.2 pwd
/home/rob/Software/blender-3.2.1-linux-x64/3.2
(blender) ➜  3.2 ll
total 12
drwxr-xr-x 8 rob rob 4096 Jul  6 11:01 datafiles
drwxr-xr-x 5 rob rob 4096 Jul  6 11:01 _python
lrwxrwxrwx 1 rob rob   33 Jul 24 12:22 python -> /home/rob/miniconda3/envs/blender
drwxr-xr-x 9 rob rob 4096 Jul  6 11:01 scripts
```

Hope this helps.

**Comment by GeraZerbetto** [comment]
I had the same issue and nothing in this thread have worked for me. I solved it by upgrading python from 3.8 to 3.10 (same version as Blender's) and reinstalling all the needed packages. Hope this helps!
Regards and thanks!

**Comment by MiekeRoth** [comment]
@GeraZebetto thanks, I’m doing that tomorrow

**Comment by markahix** [comment]
Having the same issue as others.

Ubuntu 20.04
Blender 3.3.0
Molecular Nodes 0.13.0

I've got MDAnalysis installed in a conda environment I built specifically for Blender, and have the path correct in the Molecular Nodes plugin settings.  I have tried the above scripts and suggestions with no luck.

**Comment by markahix** [comment]
Sorry, should've been more clear!  It's not recognizing MDAnalysis despite having the path correct.  When I open an interactive python session and import MDAnalysis, it works fine.

**Comment by BradyAJohnston** [comment]
When you click the `Check MDAnalysis Installation` button, is there an error that pops up? Can you open the console by going `Window` -> `Toggle System Console` and see what error messages are popping up in there. WIthout specific error messages I can't really help.

**Comment by markahix** [comment]
Clicking the red "Check Installation" button returns a hovering tooltip under the mouse which reads: "ReportError:  MDAnalysis Unavailable".

In the python console under the scripting tab, I get only "MDAnalysis Unavailable"

Somehow, I feel like Blender isn't giving me much to go on with errors.

**Comment by BradyAJohnston** [comment]
I am unsure how to get the proper console output on Linux, can you please either go `Window` -> `Toggle System Console` or if that isn't an option (isn't on Mac, is on Windows) start Blender from a terminal which will enable the terminal output and more verbose error logging. Then try the check installation button and report the errors that you get.

**Comment by asandweiss** [comment]
Was having the same issue as many people above, I tried installing on both python and conda, no luck. Tried all the tricks listed, still said "MDAnalysis Unavailable." Then I restarted my computer...
The path that previously didn't work was still in Blender, I clicked check and it worked.
Sometimes it's a good ol' fashion reset that does the trick. Anyway, many thanks!

**Comment by recombinatrix** [comment]
Right.  Same issue as above.  It's proving quite intractable.

Running on an M1 mac with MacOS 12.2.1
Download blender for apple silicon (blender-3.3.1-macos-arm64)
Already have a working conda environment with mdanalysis in python 3.9
Set up a conda environment with MDAnalysis for Blender:

```
conda create -n Blender python=3.10
conda install numpy pandas cython MDAnalysis
```

Activate this environment.  Test this MDAnalysis installation works by rerunning some RMSD calculations on large trajectories; it works fine.

Launch blender from terminal.
Install molecular_nodes_0.13.0
Install Atomium
Try to provide path to MDAnalysis installation.  I try the path to the blender specific installtion, and to my working python  nine installation, with and without training slashes.  All fail;  only terminal message is `Error: MDAnalysis Unavaiable`

Run the first diagnostic script you provided above:

```
import sys
import os
import site

# path to python.exe
python_exe = os.path.realpath(sys.executable)


def verify_user_sitepackages(package_location):

    if os.path.exists(package_location) and package_location not in sys.path:
        sys.path.append(package_location)


verify_user_sitepackages(site.getusersitepackages())
verify_user_sitepackages('/.local/lib/python3.10/site-packages')


import MDAnalysis as mda
print("succesfully imported")
```

output:

```
Error: Python: Traceback (most recent call last):
  File "/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/util.py", line 219, in <module>
    from ._cutil import unique_int_1d
ImportError: dlopen(/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/_cutil.cpython-310-darwin.so, 0x0002): tried: '/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/_cutil.cpython-310-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/lib/_cutil.cpython-310-darwin.so' (no such file)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Text", line 19, in <module>
  File "/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/__init__.py", line 191, in <module>
    from .lib import log
  File "/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/__init__.py", line 34, in <module>
    from . import transformations
  File "/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/transformations.py", line 173, in <module>
    from .mdamath import angle as vecangle
  File "/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/mdamath.py", line 63, in <module>
    from . import util
  File "/Users/ada/opt/anaconda3/envs/Blender/lib/python3.10/site-packages/MDAnalysis/lib/util.py", line 221, in <module>
    raise ImportError("MDAnalysis not installed properly. "
ImportError: MDAnalysis not installed properly. This can happen if your C extensions have not been built.
```

It seems like there's a problem with cython.  That is extremely weird to me, given that MDAnalysis works normally for actual MD Analysis, which I would expect to require c libraries.  I wonder if Blender python isn't playing well with apple silicon + conda

**Comment by recombinatrix** [comment]
Additionally, there's a typo in the terminal error message when I try and fail to provide a path to MDAnalysis in the MolecularNodes preferences; the error message in the terminal and the error report reads `Error: MDAnalysis Unavaiable` (no L in unavailable).

That's quite trivial but I mention it in case it's useful for identifying source of the problem.

(Please never read my code it is full of typos)

**Comment by BradyAJohnston** [comment]
Thanks for the error reports! I'm currently in the middle of a complete re-write of Molecular Nodes which will hopefully help with some of the issues that people are getting (and fix a bunch of the typos ha).

I haven't got access to apple silicon (one day..). Blender does seem to be missing some C headers which is why it's hard / impossible to install MDAnalysis into Blender's python natively - although so far the issue has been solved with the linking to the external conda environment.

Are you able to trial this on another computer by any chance? If it succeeds on another computer, maybe a reinstall of Blender? I currently can't think other methods for troubleshooting.

I know others have installed Molecular Nodes on M1 macs but I don't know if anyone has done the MDAnalysis linking.

**Comment by recombinatrix** [comment]
So, I did a fresh install of Blender today on my apple silicon mac and my ubuntu desktop.  A bit of tweaking got ubuntu working, but so far the mac has refused to work across three separate installs.

I'll see if I can come up with a solution following something silimar to RobertArbon's comment above, but otherwise I suspect this one might need to wait a bit.

Happy to run tests and/or generate error reports on apple silicon if that's useful, and I appreciate all your work.

**Comment by BradyAJohnston** [comment]
Thanks! Glad at least the desktop is working. 

I'll hopefully have a the refactored version ready for public testing next week, and if the issues aren't fixed then we can do some deeper dives into troubleshooting.

**Comment by famrein** [comment]
Hello, I'm having trouble installing this on Windows. I could install MDAnalysis at first, but then I couldn't get it to work in Blender. So, I reinstalled python, and pip, and everything. Now im getting this error when doing  `pip install --upgrade MDAnalysis`:


`Collecting MDAnalysis
  Using cached MDAnalysis-2.3.0.tar.gz (3.7 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 4294967295
  ╰─> [6 lines of output]
      Attempting to autodetect OpenMP support... Did not detect OpenMP support.
      No openmp compatible compiler found default to serial build.
      Will attempt to use Cython.
      *** package "numpy" not found ***
      MDAnalysis requires a version of NumPy (>=1.20.0), even for setup.
      Please get it from http://numpy.scipy.org/ or install it through your package manager.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 4294967295
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.`

I do have numpy installed:

`PS C:\Users\Fernando> pip show numpy
Name: numpy
Version: 1.23.5
Summary: NumPy is the fundamental package for array computing with Python.
Home-page: https://www.numpy.org
Author: Travis E. Oliphant et al.
Author-email:
License: BSD
Location: C:\Users\Fernando\AppData\Local\Programs\Python\Python311\Lib\site-packages
Requires:
Required-by:`

I know my computer has always had trouble with libraries because at some point I messed up the paths or something, but now I completely cleaned Python and reinstalled it and I'm still having trouble.

**Comment by BradyAJohnston** [comment]
I have now published [MolecularNodes 2.0](https://github.com/BradyAJohnston/MolecularNodes/releases) which should hopefully fix issues that people are having with installation on MDAnalysis (should now be a 1-click install). 

At least it has fixed it on my machines which would otherwise require multi-step installation. If people in this thread who have had trouble could test it out and let me know if the installation goes smoothly, that would be very much appreciated!

**Comment by BradyAJohnston** [comment]
I am going to be closing this issue, as it seems to be mostly resolved for Linux users from what I can tell, there is now only an issue with M1 / M2 macs which is tracked in #108
